zipline/lib/adjusted_array.py

```diff
@@ -165,6 +166,46 @@ def __init__(self, data, adjustments, missing_value):
         self.adjustments = adjustments
         self.missing_value = missing_value
 
+    def append_adjustments(self, adjustments_to_append, append_back):
+        """
+        Adds ``adjustments_to_append`` to existing adjustments, handling
+        index collisions according to ``append_back``.
+
+        Parameters
+        ----------
+        adjustments_to_append : dict[int -> list[Adjustment]]
+            The mapping of row indices to lists of adjustments that should be
+            appended to existing adjustments.
+        append_back : bool
```

Comment:
Related to the note above, `append_back` reads slightly off to me. One option would be to call this `append` and have `append=False` mean "prepend", though that's somewhat confusing b/c it could be read as "don't append at all". Another option might do something like `np.searchsorted` and have `side='front'` or `side='back'`.